core-bundle/tests/DependencyInjection/Compiler/RegisterFragmentsPassTest.php

```diff
@@ -177,6 +180,72 @@ public function testFailsIfThereIsNoPreHandlersDefinition(): void
         $pass->process($container);
     }
 
+    public function testRegistersTheGlobalsMapListener(): void
+    {
+        $contentController = new Definition('App\Fragments\Text');
+        $contentController->addTag('contao.content_element', ['category' => 'content']);
+
+        $container = $this->getContainerWithFragmentServices();
+        $container->setDefinition('app.fragments.content_controller', $contentController);
+
+        (new ResolveClassPass())->process($container);
+
+        $pass = new RegisterFragmentsPass(ContentElementReference::TAG_NAME, 'TL_CTE', ContentProxy::class);
+        $pass->process($container);
+
+        $definition = null;
+
+        foreach ($container->getDefinitions() as $def) {
+            if (GlobalsMapListener::class === $def->getClass()) {
+                $definition = $def;
+                break;
+            }
+        }
+
+        $this->assertNotNull($definition);
+
+        $this->assertSame(
+            [
+                'contao.hook' => [
+                    [
+                        'hook' => 'initializeSystem',
+                        'priority' => 255,
+                    ],
+                ],
+            ],
+            $definition->getTags()
+        );
+
+        $this->assertSame(
+            [
+                'TL_CTE' => [
+                    'content' => [
+                        'text' => ContentProxy::class,
+                    ],
+                ],
+            ],
+            $definition->getArguments()[0]
+        );
+
+        $this->assertTrue($definition->isPublic());
+    }
+
+    public function testFailsToRegisterGlobalsMapListenerIfCategoryIsMissing(): void
+    {
+        $this->expectException(InvalidConfigurationException::class);
```

Comment:
You should probably move this to line 246, so it becomes clear where exactly we expect the exception to be thrown. This will also ensure that we are not accidentally catching an earlier exception (although this is very unlikely in this specific case).